lib/routes/futunn/highlights.js

```diff
@@ -1,56 +1,69 @@
 const got = require('@/utils/got');
 const cheerio = require('cheerio');
-const date_util = require('@/utils/date');
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
 
 module.exports = async (ctx) => {
     const link = `https://news.futunn.com/main`;
 
     const response = await got.get(link);
     const $ = cheerio.load(response.data);
-    // extract json from 'window._params = {...};'
-    const params = $('script:not([src])').html().trim().slice(17, -1);
+
+    const contents = $('script:not([src]):contains("window._params")');
+    const params = contents.html().trim().slice(17, -1);
 
     const newslist = JSON.parse(params).preList;
 
     const out = await Promise.all(
-        newslist.map(async (info) => {
-            const title = info.title;
-            const itemUrl = info.url;
-
-            const cache = await ctx.cache.get(itemUrl);
-            if (cache) {
-                return Promise.resolve(JSON.parse(cache));
-            }
-
-            const response = await got({
-                url: itemUrl,
-                method: 'get',
-                headers: {
-                    Referer: link,
-                },
-            });
-
-            const $ = cheerio.load(response.data);
-
-            const description = $('div.newsDetailBox div.main div.inner').html() || '原文已被删除';
-
-            const date = $('div.timeBar').text();
-
-            const single = {
-                title,
-                link: itemUrl,
-                description,
-                pubDate: date_util(date, 8),
-            };
-            ctx.cache.set(itemUrl, JSON.stringify(single));
-            return Promise.resolve(single);
-        })
+        newslist
+            .filter((info) => {
+                const url = new URL(info.url);
+                if (url.hostname !== 'news.futunn.com') {
+                    // ignore videos.futunn.com
+                    return false;
+                } else if (url.searchParams.get('src') !== '3') {
+                    // ignore src=12 redirecting to mp.weixin.qq.com
+                    return false;
+                }
+                return true;
+            })
+            .map(async (info) => {
+                const title = info.title;
+                const itemUrl = info.url;
+
+                const cache = await ctx.cache.get(itemUrl);
+                if (cache) {
+                    return Promise.resolve(JSON.parse(cache));
+                }
```

Comment:
Please use `ctx.cache.tryGet()` for caching.
Docs: https://docs.rsshub.app/joinus/quick-start.html#ti-jiao-xin-de-rsshub-gui-ze-bian-xie-jiao-ben-shi-yong-huan-cun

Reply:
Thanks. Revised to use this caching interface.